src/org/daisy/dotify/formatter/impl/row/SegmentProcessor.java

```diff
@@ -1190,26 +1186,88 @@ private String getLeftIndentWithLabel(String listLabel, int indent) {
+                            + "total available width: " + (processorContext.getAvailable() + rightMargin)
+                            + ", leader position: " + leaderPos
+                            + ", leader alignment: " + leaderManager.getCurrentLeader().getAlignment()
+                            + ", text after leader: \"" + next + btr.getTranslatedRemainder() + "\""
+                            + (rightIndent > 0 ? ", right text indent: " + rightIndent : "")
+                            + (rightMargin > 0 ? ", right margin: " + rightMargin : "")
+                            + ")"
+                        );
+                    }
+                    break;
+                case LEFT:
+                    if (available < 0) {
+                        if (lineProps.getReservedWidth() > 0) {
+                            // if width is temporarily reduced try the next line
+                            return Optional.ofNullable(flushCurrentRow());
+                        }
+                        int rightMargin = currentRow.getRightMargin().getContent().length();
+                        throw new RuntimeException(
+                            "Block is too narrow to fit the leader ("
+                            + "total available width: " + (processorContext.getAvailable() + rightMargin)
+                            + ", leader position: " + leaderPos
+                            + (rightIndent > 0 ? ", right text indent: " + rightIndent : "")
+                            + (rightMargin > 0 ? ", right margin: " + rightMargin : "")
+                            + ")"
+                        );
                     }
-                    // tabSpace has been inserted, discard the leader now
-                    leaderManager.removeLeader();
                 }
+                currentRow.text(row.getLeftIndent() + currentRow.getText() + tabSpace + next);
+                if (!tabSpace.isEmpty()) {
+                    currentRow.leaderSpace(currentRow.getLeaderSpace() + tabSpace.length());
+                }
+                // tabSpace has been inserted, discard the leader now
+                leaderManager.removeLeader();
+            }
+            // If there is no leader, just insert any content that fits on the line.
+            else if (!next.isEmpty()) {
+                currentRow.text(row.getLeftIndent() + currentRow.getText() + next);
+            }
+            // If no text fits on the line because there is no available space, this is either
+            // because the previous segment filled up the row completely, in which case the row will
+            // be flushed next, or if a new row was started it's because there is no space between
+            // the left and right margin edges for text and left and right text indent, in which
+            // case we need to abort in order to avoid an endless loop.
+            else if (available < 0 || available == 0 && contentLen == 0 && btr.hasNext()) {
```

Comment:
Just for readability perhaps you should change the two first arguments either to ```available <= 0``` or add parentheses. Even if the order you should interpret these is documented I can't by a glace remember what the case would be.

Reply:
`&&` is stronger. I'll add parentheses.

Reply:
I've added parentheses.